### PR TITLE
Build amp with Gradle and publish amp to maven central

### PR DESCRIPTION
**Pull request by rawiabenhmida**
reset master on master-second-fix

### CODE REVIEW

File: alfresco.log.2020-04-17 [hunk 1]
```diff
@@ -0,0 +1,2 @@
+2020-04-17 06:56:37,016 ERROR [com.arondor.viewer.common.document.id.DocumentIdFactory] [main] No bean named documentIdFactory can be found in application context : arender.xml
```
Comment (vierbergenlars) [inline alfresco.log.2020-04-17]: Remove this log file

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    version = "$baseVersion-SNAPSHOT"
+
+
+repositories {
+    mavenCentral()
+    alfrescoPublic()
+}
+
+
+ext.alfrescoVersion = "5.1.e"
```
Comment (vierbergenlars) [inline build.gradle]: Since you are not using this variable anywhere, please remove it

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+
+
+test {
+    useJUnitPlatform()
+}
+
+dependencies {
+    compile group: 'org.alfresco', name: 'alfresco-platform-distribution', version: '5.1.e', ext: 'pom'
```
Comment (vierbergenlars) [inline build.gradle]: I think this dependency is not necessary.
All necessary dependencies are already added by depending on `org.alfresco:alfresco-repository`

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    classifier = 'sources'
+    from sourceSets.main.allSource
+}
+
+task javadocJar(type: Jar, dependsOn: javadoc) {
+    classifier = 'javadoc'
+    from javadoc.destinationDir
+}
+
+amp {
```
Comment (vierbergenlars) [inline build.gradle]: If no configuration is necessary, this block can be removed.

File: settings.gradle [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+/*
+// To declare projects as part of a multi-project build use the 'include' method
+include 'shared'
+include 'api'
+include 'services:webservice'
+*/
+
+rootProject.name = 'Arender4Alfresco'
+include 'ARender4Alfresco - transformation engine'
```
Comment (vierbergenlars) [inline settings.gradle]: Remove this include, as that subproject does no longer exist

File: src/main/java/com/arondor/arender/alfresco/content/transformer/ARenderContentTransformerWorker.java [hunk 6]
```diff
@@ -11,56 +11,49 @@
...
     {
-        return restClient.getWeatherPerformance() != -1;
+        return restClient.getWeatherPerformance() >= 0;
     }
 
     @Override
     public String getVersionString()
     {
-        return "1.0.0";
+        return "2.0.0";
```
Comment (vierbergenlars) [inline src/main/java/com/arondor/arender/alfresco/content/transformer/ARenderContentTransformerWorker.java]: Why did you change this version string to `2.0.0`?

File: src/main/java/com/arondor/arender/alfresco/content/transformer/ContentTransformARenderRestClient.java [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+    public float getWeatherPerformance()
+    {
+        try
+        {
+            return template.getForObject(UriComponentsBuilder.fromHttpUrl(address + "weather").build().toUri(),
+                    Float.class);
+        }
+        catch (Exception e)
+        {
+            LOGGER.info("Caught exception while trying to fetch weather score, is rendition server down?");
```
Comment (vierbergenlars) [inline src/main/java/com/arondor/arender/alfresco/content/transformer/ContentTransformARenderRestClient.java]: Add the exception that was thrown here to the logged information.

That way, if something goes wrong, it's easier to debug the problem.

```suggestion
            LOGGER.log(java.util.logging.Level.INFO, "Caught exception while trying to fetch weather score, is rendition server down?", e);
```

I know this isn't your code, but I think this change would make debugging an issue here a lot easier.

File: src/main/resources/alfresco-global.properties [hunk 8]
```diff
@@ -171,3 +171,7 @@ content.transformer.arender2pdf.extensions.xltx.pdf.priority=30
 
 content.transformer.arender2pdf.extensions.zip.pdf.supported=true
 content.transformer.arender2pdf.extensions.zip.pdf.priority=30
+
```
Comment (vierbergenlars) [inline src/main/resources/alfresco-global.properties]: You have 2 copies of this file laying around: one here and one in `src/main/amp/config`.
We should only keep the one in `src/main/amp/config`, and this one should be deleted.

This file is located in the root of the jar file inside the amp, and will be ignored anyways.

File: src/main/java/com/arondor/arender/alfresco/content/transformer/ARenderContentTransformerWorker.java [hunk 9]
```diff
@@ -11,56 +11,49 @@
...
 public class ARenderContentTransformerWorker implements ContentTransformerWorker, InitializingBean
 {
-    private DocumentServiceRestClient restClient = new DocumentServiceRestClient();
+    private final ContentTransformARenderRestClient restClient = new ContentTransformARenderRestClient();
+
+    private String arenderRenditionServerAddress = null;
 
-    private String arenderRenditionServerAddress = "http://localhost:8761/";
+    public ARenderContentTransformerWorker() {
+        restClient.setAddress(arenderRenditionServerAddress);
```
Comment (vierbergenlars) [inline src/main/java/com/arondor/arender/alfresco/content/transformer/ARenderContentTransformerWorker.java]: Do not call the setter here, since the variable has not yet been set at the time the constructor is called.

File: build.gradle [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+                licenses {
+                    license {
+                        name = 'GNU Lesser General Public License v3.0'
+                        url = 'https://www.gnu.org/licenses/lgpl-3.0.txt'
+                    }
+                }
+            }
+        }
+        mavenJava(MavenPublication) {
+            artifact amp
```
Comment (vierbergenlars) [inline build.gradle:90]: Are we going to add the sources jar that we created above?

In that case, please add it as an artifact.

```suggestion
            artifact amp
            artifact sourcesJar
```

**Review by vierbergenlars — CHANGES_REQUESTED** [review]
Just some minor remarks, I did not see any major problems.

File: src/main/java/com/arondor/arender/alfresco/content/transformer/ARenderContentTransformerWorker.java (hunk 9 above)
Comment (rawiabenhmida) [reply src/main/java/com/arondor/arender/alfresco/content/transformer/ARenderContentTransformerWorker.java]: @vierbergenlars when does the variable arenderRenditionServerAddress get the value from alfresco-global.properties??  (i was thinking the first thing done is setting all variables from alfresco-global.properties and then running this code isn't?

File: src/main/java/com/arondor/arender/alfresco/content/transformer/ContentTransformARenderRestClient.java [hunk 11]
```diff
@@ -42,7 +42,7 @@ public float getWeatherPerformance()
         }
         catch (Exception e)
         {
-            LOGGER.info("Caught exception while trying to fetch weather score, is rendition server down?");
+            LOGGER.log(java.util.logging.Level.INFO, "Caught exception while trying to fetch weather score, is rendition server down?");
```
Comment (vierbergenlars) [inline src/main/java/com/arondor/arender/alfresco/content/transformer/ContentTransformARenderRestClient.java]: Missing the exception in the method call.
I switched to this syntax so we would be able to log the exception too, not just so it looks more ugly :smile: 
```suggestion
            LOGGER.log(java.util.logging.Level.INFO, "Caught exception while trying to fetch weather score, is rendition server down?", e);
```

File: src/main/java/com/arondor/arender/alfresco/content/transformer/ARenderContentTransformerWorker.java (hunk 9 above)
Comment (vierbergenlars) [reply src/main/java/com/arondor/arender/alfresco/content/transformer/ARenderContentTransformerWorker.java]: Well, the "properties" defined in the xml are set by calling the setter.
Before you can call a setter on an object, it first needs to exist, so the constructor has to be called to construct the object.

After those properties have been set, then `afterPropertiesSet()` is called by spring, and your properties have the correct value.

File: src/main/java/com/arondor/arender/alfresco/content/transformer/ContentTransformARenderRestClient.java [hunk 12]
```diff
@@ -42,13 +42,15 @@ public float getWeatherPerformance()
...
         {
-            LOGGER.log(java.util.logging.Level.INFO, "Caught exception while trying to fetch weather score, is rendition server down?");
+            LOGGER.log(java.util.logging.Level.INFO, "Caught exception while trying to fetch weather score, is rendition server down?",e);
             return -1;
         }
     }
 
     public void setAddress(String address)
     {
+        ARenderContentTransformerWorker aRenderContentTransformerWorker=new ARenderContentTransformerWorker();
```
Comment (vierbergenlars) [inline src/main/java/com/arondor/arender/alfresco/content/transformer/ContentTransformARenderRestClient.java]: You should not create this object yourself. It is created automatically by spring.

**Review by vierbergenlars — CHANGES_REQUESTED** [review]

File: src/main/amp/config/alfresco/module/com.arondor.arender.alfresco.content.transformer/alfresco-global.properties [hunk 13]
```diff
@@ -171,3 +171,4 @@ content.transformer.arender2pdf.extensions.xltx.pdf.priority=30
 
 content.transformer.arender2pdf.extensions.zip.pdf.supported=true
 content.transformer.arender2pdf.extensions.zip.pdf.priority=30
+content.transformer.arender2pdf.url=http://10.233.18.244:8761/
```
Comment (vierbergenlars) [inline src/main/amp/config/alfresco/module/com.arondor.arender.alfresco.content.transformer/alfresco-global.properties]: Do not set a URL here. It must be provided by the user from `alfresco-global.properties`, and no default should be set here

**Review by vierbergenlars — CHANGES_REQUESTED** [review]

**Review by vierbergenlars — APPROVED** [review]
Please make sure to squash all your commits when merging to master
